### ISSUE TITLE
Tab completion issue and missing clojure pieces on Ubuntu Cuttlefish

### ISSUE THREAD

**Issue by geokon-gh** [technomancy/leiningen#2491]
I'm working off a plain system: 
```
OS: Ubuntu 18.10 cosmic
Kernel: x86_64 Linux 4.18.0-11-generic
```

 I install OpenJDK 8 through `apt`

```
geokon@ux305-ubuntu:~$ apt list --installed | grep openjdk

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

openjdk-8-jdk-headless/cosmic-updates,cosmic-security,now 8u191-b12-0ubuntu0.18.10.1 amd64 [installed,automatic]
openjdk-8-jdk/cosmic-updates,cosmic-security,now 8u191-b12-0ubuntu0.18.10.1 amd64 [installed]
openjdk-8-jre-headless/cosmic-updates,cosmic-security,now 8u191-b12-0ubuntu0.18.10.1 amd64 [installed,automatic]
openjdk-8-jre/cosmic-updates,cosmic-security,now 8u191-b12-0ubuntu0.18.10.1 amd64 [installed,automatic]
```

I download the lein script and put it in `~/bin/`,  reboot and then run `lein` followed by `lein repl`. Lots of things get downloaded and things seem to be working

```
geokon@ux305-ubuntu:~/bin$ lein repl
nREPL server started on port 43853 on host 127.0.0.1 - nrepl://127.0.0.1:43853
REPL-y 0.3.7, nREPL 0.2.12
Clojure 1.8.0
OpenJDK 64-Bit Server VM 1.8.0_191-8u191-b12-0ubuntu0.18.10.1-b12
    Docs: (doc function-name-here)
          (find-doc "part-of-name-here")
  Source: (source function-name-here)
 Javadoc: (javadoc java-object-or-class-here)
    Exit: Control+D or (exit) or (quit)
 Results: Stored in vars *1, *2, *3, an exception in *e

user=> (+ 1 2)
3
```

But then I start noticing things are a bit weird and I came across how broken my setup seems to be when I was trying to get an old project of mine running: https://github.com/fn-fx/fn-fx/issues/68#issuecomment-443101537

I've narrowed down two issues:
1.
When I type in `leiningen.co` and then hit `TAB` to autocomplete I get a weird error which comes out looking like this:
```
user=> (leiningen.coNullPointerException   complete.core/fn--632/iter--624--633/fn--634 (core.clj:66)
```

2.
When I try to run `(clojure.reflect/resolve-class (.getContextClassLoader (Thread/currentThread)) 'javafx.application.Platform)` I get that `clojure.reflect` doesn't exist:
```
user=> (clojure.reflect/resolve-class (.getContextClassLoader (Thread/currentThread)) 'javafx.application.Platform)

ClassNotFoundException clojure.reflect  java.net.URLClassLoader.findClass (URLClassLoader.java:382)
```
However it's been part of Clojure since v1.3

I'm pretty new to Clojure, so I'm at a loss as to how to diagnose this problem further and to narrow down if this is a problem with my system or with `lein`. If anyone has any suggestions as to how to move forward, please let me know

(I've also tested with OpenJDK 11 with the same results)

**Comment by geokon-gh** [comment]
What does "installed manually" mean? 

I'm currently not using the version from the repository. I'm using the download script from the homepage (im pretty sure the one in the repository had the same issues as well)

And I'm not sure how to build it. Can I do that without Clojure?

**Comment by technomancy** [comment]
Oh sorry; I misread. You got your JDK from Ubuntu, but not your lein script, so this should be tracked here.

However, I can't reproduce the first problem. Can you comment out your ~/.lein/profiles.clj file if present and see if it still happens?

The second problem is not a bug; clojure.reflect must be required before its functions can be called.

**Comment by geokon-gh** [comment]
I don't have a ` ~/.lein/profiles.clj `. This is just a fresh blank install - should it still be there?

And thanks for the info about the second part. I'll forward it on

**Comment by geokon-gh** [comment]
Alright. I kinda by accident confirmed that someone else had this same behavior (if you check the issue I linked). So I'm not alone. Maybe if it happens with someone else they will chime in here :)

**Comment by glts** [comment]
This was a bug in clojure-complete, see ninjudd/clojure-complete#24. This bug has been fixed in clojure-complete 2.5, which is included in Leiningen 2.9.1.

(The other bug report is for Java 9, but I imagine that the cause is the same with the Ubuntu-packaged Java 8 here.)
